### ISSUE TITLE
Allow setting margins (insets) for watermarks

### ISSUE THREAD

**Issue by rider87** [coobird/thumbnailator#177]
Hi all,

Is there a possibility to have margin for the watermark?
Like you maybe know from Lightroom?

Please see this example... There is a space between bottom and left side...
Is this also possible?

![image](https://user-images.githubusercontent.com/9787533/134177075-b02ec183-a060-44a1-ae65-466a87c37383.png)

**Comment by rider87** [comment]
I believe it´s just to adjust this code:
![image](https://user-images.githubusercontent.com/9787533/134179280-369a2c52-a241-4408-9c13-6de803d08f2a.png)

Currently only one value is there... 
Can we do margin for 
- top
- bottom
- right
- left

**Comment by coobird** [comment]
Are you asking whether it's possible to have top/bottom/left/right margins to be set independently? If that's the case, as you found in the code, right now that's not possible.

**Comment by rider87** [comment]
yes, correct.... currently I can see to set only one value for all (top/bottom/left/right)...

Can this be implemented?
I believe there needs just to add 3 addtional parameters to the function...

**Comment by coobird** [comment]
I'm hesitant to add yet another overload for the `watermark` method to the Thumbnailator API, especially one that has 7 arguments. That said, I'll keep this issue open as this sounds like a valid use case.
